File: midna/src/main/java/com/pel/midna/model/Tournament.java
package com.pel.midna.model;

import java.sql.Timestamp;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

/**
 * User: bharat
 * Date: 8/10/21
 * Time: 1:23 PM
 */
public class Tournament {

    public Integer id;
    public String name;
    public String desc;
    public String game;
    public String type;
    public String division;
    public Timestamp registrationStart;
    public Timestamp registrationEnd;
    public Timestamp seasonStart;
    public Timestamp seasonEnd;
    public Timestamp playoffStart;
    public Timestamp createdAt;
    public Timestamp updatedAt;

    public List<Map<String, Object>> teams = new ArrayList<>();
}

File: lib/models/tournament.dart
import 'package:pel_portal/utils/config.dart';

class Tournament {
  int? id;
  String? name;
  String? desc;
  String? game = "VALORANT";
  String? type = "HIGH_SCHOOL";
  String? division;

  DateTime? registrationStart;
  DateTime? registrationEnd;
  DateTime? seasonStart;
  DateTime? seasonEnd;
  DateTime? playoffStart;

  DateTime? updatedAt;
  DateTime? createdAt;

  Tournament();

  Tournament.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    name = json['name'];
    desc = json['desc'];
    game = json['game'];
    type = json['type'];
    division = json['division'];
    registrationStart = DateTime.tryParse(json['registrationStart']);
    registrationEnd = DateTime.tryParse(json['registrationEnd']);
    seasonStart = DateTime.tryParse(json['seasonStart']);
    seasonEnd = DateTime.tryParse(json['seasonEnd']);
    playoffStart = DateTime.tryParse(json['playoffStart']);
    createdAt = DateTime.tryParse(json['createdAt']);
    updatedAt = DateTime.tryParse(json['updatedAt']);
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'name': name ?? "null",
    'desc': desc ?? "null",
    'game': game ?? "null",
    'type': type ?? "null",
    'division': division ?? "null",
    'registrationStart': registrationStart != null ? registrationStart.toString() : "null",
    'registrationEnd': registrationEnd != null ? registrationEnd.toString() : "null",
    'seasonStart': seasonStart != null ? seasonStart.toString() : "null",
    'seasonEnd': seasonEnd != null ? seasonEnd.toString() : "null",
    'playoffStart': playoffStart != null ? playoffStart.toString() : "null",
    'createdAt': createdAt != null ? createdAt.toString() : "null",
    'updatedAt': updatedAt != null ? updatedAt.toString() : "null",
  };
}